crates/legacy/src/service.rs

```diff
@@ -387,6 +397,56 @@ impl<Types: NodeType> GlobalState<Types> {
         .await
     }
 
+    // get transaction status
+    // return one of "pending", "sequenced", "rejected" or "unknown"
+    pub async fn txn_status(
+        &self,
+        txn_hash: Commitment<<Types as NodeType>::Transaction>,
+    ) -> Result<TransactionStatus, BuildError> {
+        if let Some(status) = self.tx_status.write_arc().await.get(&txn_hash) {
+            Ok(status.clone())
+        } else {
+            Ok(TransactionStatus::Unknown)
+        }
+    }
+
+    pub async fn set_txn_status(
+        &mut self,
+        txn_hash: Commitment<<Types as NodeType>::Transaction>,
+        txn_status: TransactionStatus,
+    ) -> Result<(), BuildError> {
+        if self.tx_status.write_arc().await.contains(&txn_hash) {
+            let mut write_guard = self.tx_status.write_arc().await;
```

Comment:
Will these two `write_arc()` calls potentially cause deadlocks? Also, can they just be read locks?

Reply:
Good catch! updated in [d94a136](https://github.com/EspressoSystems/marketplace-builder-core/pull/236/commits/d94a1361b253d10ceb3b7d1ec9865f910f645bff).